ext/soap/soap.c

```diff
@@ -1166,8 +1165,8 @@ PHP_METHOD(SoapServer, addFunction)
 			}
 			service->soap_functions.functions_all = TRUE;
 		} else {
-			php_error_docref(NULL, E_WARNING, "Invalid value passed");
-			return;
+			zend_argument_value_error(1, "must be SOAP_FUNCTIONS_ALL when an integer is passed");
+			RETURN_THROWS();
```

Comment:
Shouldn't there be one more else below to handle a completely invalid type?

Reply:
Yeah, I missed that, thanks!